src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs

```diff
@@ -464,6 +464,130 @@ public ResultWrapper<IReadOnlyList<SimulateBlockResult<GethLikeTxTrace>>> debug_
+        }
+
+        // Smart Path Bifurcation: Mixed-Mode Handling Trade-offs
+        // 
+        // DESIGN DECISION: All-or-Nothing Override Processing
+        // We route ALL bundles to simulation infrastructure if ANY bundle contains overrides.
+        // This "all-or-nothing" approach was chosen for several reasons:
+        //
+        // PERFORMANCE vs COMPLEXITY TRADE-OFF:
+        // - Simple Path (DebugBridge.GetBundleTraces): ~2-3x faster, minimal memory allocation
+        // - Override Path (SimulateTxExecutor): Full simulation infrastructure, higher overhead
+        //
+        // CURRENT APPROACH BENEFITS:
+        // - Guaranteed execution consistency across all bundles in a single request
+        // - Simplified error handling and timeout management
+        // - No complex result merging or ordering issues
+        // - Clear, predictable performance characteristics
+        if (bundles.Any(b => b.BlockOverride is not null || b.StateOverrides != null))
+        {
+            return TraceCallManyWithOverrides(bundles, blockParameter, options, header);
+        }
+
+        foreach (var bundle in bundles)
+        {
+            foreach (var call in bundle.Transactions)
+            {
+                call.Gas ??= header.GasLimit;
+                call.EnsureDefaults(jsonRpcConfig.GasCap);
+            }
+        }
+
+        using CancellationTokenSource timeout = BuildTimeoutCancellationTokenSource();
+        CancellationToken cancellationToken = timeout.Token;
+
+        IReadOnlyList<GethLikeTxTrace[]> bundleTraces = debugBridge.GetBundleTraces(bundles, blockParameter, cancellationToken, options)
+            .Select(bundleTrace => bundleTrace.ToArray())
+            .ToList();
+
+        if (_logger.IsTrace)
+        {
+            int totalTransactions = bundles.Sum(b => b.Transactions?.Length ?? 0);
+            _logger.Trace($"{nameof(debug_traceCallMany)} completed: {bundles.Length} bundles, {totalTransactions} transactions via simple path");
+        }
+
+        return ResultWrapper<IReadOnlyList<GethLikeTxTrace[]>>.Success(bundleTraces);
+    }
+
+    private ResultWrapper<IReadOnlyList<GethLikeTxTrace[]>> TraceCallManyWithOverrides(TransactionBundle[] bundles, BlockParameter blockParameter, GethTraceOptions? options, BlockHeader header)
+    {
+        var simulatePayload = new SimulatePayload<TransactionForRpc>
+        {
+            BlockStateCalls = bundles.Select(bundle =>
+            {
+                foreach (var call in bundle.Transactions)
+                {
+                    call.Gas ??= header.GasLimit;
+                    call.EnsureDefaults(jsonRpcConfig.GasCap);
+                }
+
+                return new BlockStateCall<TransactionForRpc>
+                {
+                    BlockOverrides = bundle.BlockOverride,
+                    StateOverrides = bundle.StateOverrides,
+                    Calls = bundle.Transactions
+                };
+            }).ToList()
```

Comment:
We could potentially change `BlockStateCalls` to `IReadOnlyList` and go with ArrayPoolList here to avoid collection allocation

But that is more advanced trick, so feel free to ignore for now